### ISSUE TITLE
Cannot use namespace 'GraphiQL' as a type

### ISSUE THREAD

**Issue by benjie** [graphql/graphiql#2466]
The [generated code for GraphiQL](https://unpkg.com/graphiql@1.9.4/dist/components/GraphiQL.d.ts) now exports a function and a namespace:

```ts
export declare function GraphiQL({ dangerouslyAssumeSchemaIsValid, docExplorerOpen, getDefaultFieldNames, inputValueDeprecation, introspectionQueryName, maxHistoryLength, onCopyQuery, onToggleHistory, onToggleDocs, storage, schema, schemaDescription, ...props }: GraphiQLProps): JSX.Element;
export declare namespace GraphiQL {
    var formatResult: (result: any) => string;
    var formatError: (error: any) => string;
    var Logo: typeof GraphiQLLogo;
    var Toolbar: typeof GraphiQLToolbar;
    var Footer: typeof GraphiQLFooter;
    var QueryEditor: typeof import("./QueryEditor").QueryEditor;
    var VariableEditor: typeof import("./VariableEditor").VariableEditor;
    var HeaderEditor: typeof import("./HeaderEditor").HeaderEditor;
    var ResultViewer: typeof import("./ResultViewer").ResultViewer;
    var Button: typeof import("./ToolbarButton").ToolbarButton;
    var ToolbarButton: typeof import("./ToolbarButton").ToolbarButton;
    var Group: typeof ToolbarGroup;
    var Menu: typeof ToolbarMenu;
    var MenuItem: React.FC<{
        onSelect: () => void;
        title: string;
        label: string;
    }>;
}
```

I'm not good enough at TypeScript to know _why_ it's doing this. But it breaks existing code that allows passing around a `graphiql: GraphiQL` ref. Could someone expand on the reason behind this change? Was it from converting from a class component to a functional one in https://github.com/graphql/graphiql/pull/2370?

```ts
import type { GraphiQL } from 'graphiql';
```

**Comment by thomasheyenbrock** [comment]
Thanks for reporting this @benjie!

Your hunch is correct, the `GraphiQL` export is a function component since `1.9.0` and not a class component anymore. That means you can't use the export as type anymore.

```ts
// Good in <1.9.0, fails in >=1.9.0
import { GraphiQL } from "graphiql";

type Foo = { graphiql: GraphiQL }; 
```

This is a breaking change that we did not notice. I'm gonna roll this back to be non-breaking (I think we can just wrap the top-level functional component in another class component for now).

As a sidenote: We're rewriting everything to modern React (functional components, contexts & hooks), so you won't be able to pass a ref to the main `GraphiQL` component in the next major version anymore. May I ask what your specific use-cases are for this? (So that we can make sure there's an alternative way to solve them in v2 😉 )

**Comment by benjie** [comment]
It’s for the new PostGraphiQL; the old one is here: https://github.com/graphile/postgraphile/blob/main/postgraphiql/src/components/PostGraphiQL.js the new one has been modernised to use functional components, hooks and context ;) but isn’t OSS yet

**Comment by thomasheyenbrock** [comment]
@benjie the fix was published earlier today with 1.9.6
